pip/commands/list.py

```diff
@@ -91,6 +91,13 @@ def __init__(self, *args, **kw):
             help="Align package names and versions into vertical columns.",
         )
 
+        cmd_opts.add_option(
+            '--leaves',
+            action='store_true',
+            dest='leaves',
+            help="List packages that are not dependencies of installed packages.",
```

Comment:
> Surely the packages that aren't depended on by anything else are [4, 5, 6]? And that's what I thought was the intent, as if I did pip install 4 5 6 I'd get 1, 2 and 3 pulled in as dependencies, which (if I understand the explanation of what's needed from #876) is what was wanted.

OK, now I see it, thanks. I will fix the code.

> Having said this, I can't personally think of a case where I'd want this (except maybe as the nearest I can get to what I actually care about, which is "give me all the packages I installed explicitly, but leave out any that were only pulled in as dependencies", which pip can't tell you as it doesn't retain that information)

The use case I have in mind is "list of packages you can delete without breaking other packages."